assets/chat/js/autocomplete.js

```diff
@@ -197,6 +197,15 @@ class ChatAutoComplete {
         this.selected = -1
         this.results = []
         this.criteria = criteria
+
+        //for emote suffixes
+        if (criteria.orig.includes(' :') && criteria.word.startsWith(':')) {
+            const last_colon = criteria.orig.lastIndexOf(' :')
+            const l = criteria.orig.length
+            //assumption: the last occurrance of " :" is ment to be a emote suffix
+            criteria.orig = criteria.orig.substring(0, last_colon) + ':' + criteria.orig.substring(last_colon + 2, l)
```

Comment:
I don't think we need to get `criteria.orig.length` at all. Omitting the second argument to `substring()` will already go to the end of the string.